unstructured_ingest/embed/bedrock.py

```diff
@@ -85,3 +94,78 @@ def embed_documents(self, elements: list[dict]) -> list[dict]:
         embeddings = [self.embed_query(query=e.get("text", "")) for e in elements]
         elements_with_embeddings = self._add_embeddings_to_elements(elements, embeddings)
         return elements_with_embeddings
+
+
+class AsyncBedrockEmbeddingConfig(EmbeddingConfig):
+    aws_access_key_id: SecretStr
+    aws_secret_access_key: SecretStr
+    region_name: str = "us-west-2"
+    embed_model_name: str = Field(default="amazon.titan-embed-text-v1", alias="model_name")
+
+    @requires_dependencies(
+        ["aioboto3"],
+        extras="bedrock",
+    )
+    @asynccontextmanager
+    async def get_client(self) -> AsyncContextManager["AsyncBedrockClient"]:
+        # delay import only when needed
+        import aioboto3
+
+        session = aioboto3.Session()
+        async with session.client(
+            "bedrock-runtime",
+            aws_access_key_id=self.aws_access_key_id.get_secret_value(),
+            aws_secret_access_key=self.aws_secret_access_key.get_secret_value(),
+            region_name=self.region_name,
+        ) as aws_bedrock:
+            yield aws_bedrock
```

Comment:
They're not always the same, which is why I wanted to support having custom configs between both implementations.

Reply:
I don't see where they are not the same, as far as I see:

- bedrock - both implementations have the same fields with the same defaults
- openai - same as above
- mixedbreadai - inherits sync config, no additional fields
- octoai - same as above
- voyageai -  same as above
- togetherai - inherits sync config but also adds the same fields as sync config
- vertexai - doesn't apply as there's one client for both sync, async

I could be misreading somewhere so please point me out if I do but if this is correct then unless we have a reason to expect async specific configuration in the future then I think this suggestion makes sense.

And if not that I think we should at least be consistent with how we define those async config classses because it's done in 3 different ways to reach the same goal.

Reply:
The method needs to be async because it's using the async generator from the `session.client()` call.